### ISSUE TITLE
New plugin dll could not be read

### ISSUE THREAD

**Issue by endragor** [OurMachinery/themachinery-public#537]
The Machinery 2021.3
Windows 10 Pro Version 1909

Very often after recompiling a plugin and returning focus back to The Machinery window, an error like this occurs:
`plugin: tm_gameplay_sample_third_person.dll could not be read`

Manually respecifying the plugin path works for reloading it, but it's still a bit annoying - you have to spot the error and then do some manual stuff to make the new version work. Would be nice if the reload system worked a bit more reliably.

I had the same issue on The Machinery 2021.2a.

**Comment by niklas-ourmachinery** [comment]
I have seen this too... but for me it usually succeeds to load the plugin after first failing a few times. I think what happens is that it tries to reload the plugin before it has been fully written out by the compile action.

I'll see if I can make it more reliable. Can you add a picture of what the log output looks like when you get this problem?

**Comment by endragor** [comment]
For me, it doesn't try to reload it again after a failure. So I have to manually uncheck and check again "Import When Changed" to force a reload. Here is a screenshot:

![the_machinery_console](https://user-images.githubusercontent.com/2417849/113588234-121c4e00-965a-11eb-94b7-fb747c5fb1fc.png)

Interestingly, when I tried to reproduce now by just recompiling a slightly different code many times, the issue occurred much more rarely than yesterday, when I was actually actively working with the editor. It felt like it was happening more than 50% of the time yesterday, so it was pretty annoying. Also, I didn't spot the error every time, sometimes seeing the old behavior and thinking I fixed something wrong only to later realize that the reload didn't work and everything is actually fine with the code...
